downstairs/src/extent_inner_sqlite.rs

```diff
@@ -508,6 +505,77 @@ impl ExtentInner for SqliteInner {
 }
 
 impl SqliteInner {
+    /// Converts to a raw file for use with `RawInner`
+    ///
+    /// Returns the metadata and context slots, which should be positioned
+    /// directly after the raw block data in memory.
+    pub fn export_meta_and_context(
+        &mut self,
+    ) -> Result<Vec<u8>, CrucibleError> {
+        // Check whether we need to rehash.  This is theoretically represented
+        // by the `dirty` bit, but we're being _somewhat_ paranoid and manually
+        // forcing a rehash if any blocks have multiple contexts stored.
+        let ctxs = self.get_block_contexts(0, self.extent_size.value)?;
+        let needs_rehash = ctxs.iter().any(|c| c.len() > 1);
```

Comment:
Feedback requested: is this paranoia justified, or should I just call `self.fully_rehash_and_clean_all_stale_contexts(false)?` and trust the dirty flag?

Reply:
I'd say this is justified - `sqlite` could have a bug that creates this edge case.